doc/user/content/serve-results/snowflake.md

```diff
@@ -0,0 +1,255 @@
+or a role with the [global `CREATE INTEGRATION` privilege](https://docs.snowflake.com/en/user-guide/security-access-control-privileges#global-privileges-account-level-privileges)
+can execute this step.
+{{< /note >}}
+
+1. In [Snowsight](https://app.snowflake.com/), or your preferred SQL client
+connected to Snowflake, create a [storage integration](https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration),
+replacing `<role>` with the name of the role you created in the previous step:
+
+   ```sql
+   CREATE STORAGE INTEGRATION S3_int
+     TYPE = EXTERNAL_STAGE
+     STORAGE_PROVIDER = 'S3'
+     ENABLED = TRUE
+     STORAGE_AWS_ROLE_ARN = 'arn:aws:iam::001234567890:role/<role>'
+     STORAGE_ALLOWED_LOCATIONS = ('*');
+   ```
+
+1. Retrieve the IAM principal for your Snowflake account using the
+   [`DESC INTEGRATION`](https://docs.snowflake.com/en/sql-reference/sql/desc-integration)
+   command:
+
+   ```sql
+   DESC INTEGRATION s3_int;
+   ```
+
+   Note down the values for the `STORAGE_AWS_IAM_USER_ARN` and
+   `STORAGE_AWS_EXTERNAL_ID` properties. You will need them in the next step to
+   update the Snowflake trust policy attached to your S3 bucket.
+
+### Update the IAM policy
+
+1. In your AWS account, find the IAM role you created in [Create an IAM role](#create-an-iam-role)
+   and, under **Trust relationships**, click **Edit trust policy**. Use the values
+   for the `STORAGE_AWS_IAM_USER_ARN` and `STORAGE_AWS_EXTERNAL_ID` properties
+   from the previous step to update the trust policy's `Principal` and
+   `ExternalId`, then click **Update policy**.
+
+## Step 3. Create a Snowflake external stage
+
+Back in Snowflake, create an [external stage](https://docs.snowflake.com/en/user-guide/data-load-S3-create-stage#external-stages) that uses the storage integration above and references your S3 bucket.
+
+```sql
+CREATE STAGE s3_stage
+  STORAGE_INTEGRATION = s3_int
+  URL = 's3://<bucket>/<prefix>/';
+```
+
+{{< note >}}
+To create a stage that uses a storage integration, the active user must have a
+role with the [`CREATE STAGE` privilege](https://docs.snowflake.com/en/sql-reference/sql/create-stage)
+for the active schema, as well as the [`USAGE` privilege](https://docs.snowflake.com/en/sql-reference/sql/grant-privilege#syntax)
+on the relevant storage integration.
+{{< /note >}}
+
+## Step 4. Import data into Snowflake
+
+To import the data stored in S3 into Snowflake, you can then create a table and
+use the [`COPY INTO`](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table)
+command to load it from the external stage.
+
+{{< tabs >}}
+{{< tab "Parquet">}}
+
+Create a table with a single column of type [`VARIANT`](https://docs.snowflake.com/en/sql-reference/data-types-semistructured#variant):
+
+```sql
+CREATE TABLE s3_table_parquet (
+    col VARIANT
+);
+```
+
+Use `COPY INTO` to load the data into the table:
+
+```sql
+COPY INTO s3_table_parquet
```

Comment:
I don't think this is idempotent. Should we suggest something like this?
```sql
COPY INTO s3_table_parquet
FROM @s3_stage
FILE_FORMAT = (TYPE = 'PARQUET')
FILES = (SELECT METADATA$FILENAME FROM @s3_stage WHERE METADATA$FILENAME NOT IN (
  SELECT DISTINCT METADATA$FILENAME
  FROM s3_table_parquet
));
```

Reply:
I'll take a closer look at their other options to integrate data from S3 (_i.e._ external tables, Snowpipe). Wish that finding best practices didn't take this much digging. 🫠